File: Java/1726. Tuple with Same Product.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        Map<Integer, Integer> products = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int n1 = nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                int n2 = nums[j];
                products.put(n1 * n2, products.getOrDefault(n1 * n2, 0) + 1);
            }
        }

        int ans = 0;
        for (int v: products.values()) {
            ans += 4 * v * (v - 1);
        }

        return ans;
    }
}

File: Python3/1726. Tuple with Same Product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        products = defaultdict(int)
        ans = 0

        for i, n1 in enumerate(nums):
            for j, n2 in enumerate(nums[i + 1:], i + 1):
                products[n1 * n2] += 1

        for v in products.values():
            if v > 1:
                ans += 4 * v * (v - 1)

        return ans